tests/test_androidmodule_ctypes_finder.py

```diff
@@ -0,0 +1,131 @@
+
+import mock
+from mock import MagicMock
+import os
+import shutil
+import sys
+import tempfile
+
+sys.modules["jnius"] = MagicMock()  # fake jnius used later
+
+
+# Import the tested android._ctypes_library_finder module,
+# making sure android._android won't crash us!
+# (since android._android is android-only / not compilable on desktop)
+android_module_folder = os.path.abspath(os.path.join(
+    os.path.dirname(__file__),
+    "..", "pythonforandroid", "recipes", "android", "src"
+))
+sys.path.insert(0, android_module_folder)
+sys.modules['android._android'] = MagicMock()
+import android._ctypes_library_finder
+sys.path.remove(android_module_folder)
+
+
+def _get_autoclass_with_fake_activity(returning_test_path="/testpath",
+                                      return_as_service=False):
+    """
+    Helper functions to mock autoclass() from jnius.
+
+    This will return a mocked activity "org.kivy.android.PythonActivity"
+    which supports .getApplicationContext().getPackagemanager(),
+    with that returned object supporting .getApplicationInfo()
+    for PackageManager's GET_SHARED_LIBRARY_FILES info.
+
+    (That is the functionality the android._ctypes_library_finder module
+     uses to scan the application-specific library directories)
+    """
+
+    # This is a closure to bind the value of "returning_test_path":
+    def autoclass_returning_graphical_activity(name):
+        activity = MagicMock()
+        activity.getApplicationContext().getPackageName = (
+             lambda: "com.example.testpackage"
+        )
+        activity.getApplicationContext().getPackageManager().\
+            getApplicationInfo().nativeLibraryDir = returning_test_path
+        activity_class = MagicMock()
+
+        if name == "org.kivy.android.PythonActivity" and \
+                not return_as_service:
+            activity_class.mService = None
+            activity_class.mActivity = activity
+            return activity_class
+        elif name == "org.kivy.android.PythonService" and return_as_service:
+            activity_class.mActivity = None
+            activity_class.mService = activity
+            return activity_class
+        elif name == "android.content.pm.PackageManager":
+            manager_class = MagicMock()
+            manager_class.GET_SHARED_LIBRARY_FILES = 1024
```

Comment:
mocking explicitly is like documenting. Would the test fail if you remove that line? If not then it means it's not mandatory to prove the point of the interface you're unit testing

Reply:
I added a check for this now in the updated test (the only part still involving a small `if` :grimacing: lol)